src/lib/ledger_catchup/README.md

```diff
@@ -0,0 +1,39 @@
+
+- catchup scheduler: the scheduler is not a part of catchup, however it controls the logic of when catchup should be invoked. when a block that has no parent is received, it gets passed to the scheduler, which then usually waits some time to see if the parent gets gossiped, and if it hasn't been, then the scheduler invokes a catchup job. 
+
+- normal and super catchup: There's two kinds of catchup in the code, normal and super catchup.  They both do the same thing, normal catchup is simply the older deprecated version.  At this moment, by default super catchup is used.  there is a mina flag `--no-super-catchup` which instructs the daemon to use the older normal-catchup, though few people do that.  the already deprecated “normal” catchup will be removed at some point and then “super” catchup will become the only catchup.  everything in this doc describes super catchup, which at this point is more normal than normal catchup.
+
+### Catchup specifics
+
+- super catchup uses a data structure called `full_catchup_tree`. each node of this tree basically represents a block that needs to be downloaded, verified, validated, and turned into a breadcrumb and added to the transition frontier (arguably it would be better to name them "jobs" instead of "nodes"). each job goes through those phases.
+    - the catchup tree is a trees of hashes of blocks, which are in a pending state and need to be downloaded and processed into the transition frontier. (it's actually a forest and not one tree, but this is just because the scheduler passes to catchup a bunch of blocks in a tree minus their common root, so it becomes a forest)
+- within the `run` function of super catchup, first it sets up a number of dependencies and creates a few aux functions.
+    - there is a very important interior function defined within the super_catchup run function called `run_node`. `run_node` is a state machine which puts each node (ie a job) of the catchup tree through it's phases. it takes a node of the transition frontier, and either fully downloads it, validates it, builds it into a breadcrumb, or builds the parent into a breadcrumb. the function looks like a big match-case expression that handles all the stages of a node. it recursively calls itself to handle the subsequent stages.
+    - the downloader is also created in the beginning of the `run` function. has a list of blocks that need to be downloaded. super catchup (i believe) downloads in a more forward order towards the target.
```

Comment:
Yes, the downloader is designed to download blocks in the same order they were produced in (as opposed to normal catchup, which did it backward).